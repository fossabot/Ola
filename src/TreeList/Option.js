import React from 'react'
import {default as PT} from 'prop-types'
import cx from 'classnames'

const TreeListOption = ({ className, children, ...props }) => {
  return (
    <li className={cx('ola_treeList-option', className)} {...props}>
      {children}
    </li>
  )
}

TreeListOption.defaultProps = {
  className: null,
}

TreeListOption.propTypes = {
  /** Extra className */
  className: PT.string,
  /** Childen nodes */
  children: PT.oneOfType([
    PT.string,
    PT.arrayOf(PT.node),
    PT.node
  ]).isRequired
}

export default TreeListOption
