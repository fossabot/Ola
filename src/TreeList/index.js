import React from 'react'
import { default as PT } from 'prop-types'
import cx from 'classnames'


const TreeList = ({ className, children, ...props }) =>
  <ul className={cx('ola_treeList', className)} {...props}>
    {children}
  </ul>


TreeList.defaultProps = {
  className: null,
}

TreeList.propTypes = {
  /** Extra className */
  className: PT.string,
  /** Childen nodes */
  children: PT.oneOfType([
    PT.string,
    PT.arrayOf(PT.node),
    PT.node
  ]).isRequired,
}

export default TreeList
