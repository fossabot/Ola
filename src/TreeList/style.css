.ola_treeList {
    list-style: none;
    padding: 0;
    margin: 0;
}
.ola_treeList-option {
    border-bottom: solid 1px var(--gray-xlight);

    &:first-child {
        border-top: solid 1px var(--gray-xlight);
    }
}
.ola_treeList .ola_treeList .ola_treeList-option {
    border: none;
    padding-left: var(--size-8);
    position: relative;

    &::before {
        content: "";
        position: absolute;
        border-left: 2px solid var(--gray-light);
        border-bottom: 2px solid var(--gray-light);
        height: 50%;
        min-height: var(--size-5);
        width: var(--size-5);
        top: 0;
        left: var(--size-5);
        border-bottom-left-radius: var(--size-5) var(--size-5);
    }

    &:not(:last-child)::after {
        content: "";
        position: absolute;
        width: 0;
        height: 100%;
        border-left: 2px solid var(--gray-light);
        top: 0;
        left: var(--size-5);
    }
}
.ola_treeList-option > a {
    display: block;
    color: inherit;
    text-decoration: none;
    background-image: url('data:image/svg+xml;utf-8,<svg width="28" height="28" viewBox="0 0 28 28" xmlns="http://www.w3.org/2000/svg"><path d="M15.586 14l-5.293-5.293a1 1 0 0 1 1.414-1.414l6 6a1 1 0 0 1 0 1.414l-6 6a1 1 0 0 1-1.414-1.414L15.586 14z" fill="%23BBC2CE"></path></svg>');
    background-repeat: no-repeat;
    background-position: right var(--size-2) center;
    padding-right: var(--size-8);
    padding-left: var(--size-3);
    padding-top: var(--size-4);
    padding-bottom: var(--size-4);

    &:hover {
        background-color: var(--gray-xlight);
    }
    &:focus {
        outline: 0;
        box-shadow: none; /* This is due WordPress */
    }

    & > * {
        margin: 0;
    }
}
.ola_treeList-option .ola_treeList-option > a {
    background-image: none;
    border-radius: var(--radius);
}