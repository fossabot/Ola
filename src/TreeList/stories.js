import React from 'react'
import TreeList from './'
import Page from '../Page'
import Icon from '../Icon'
import Level from '../Level'
import Tag from '../Tag'
import TreeListOption from './Option'
import Issue from '../Issue'

export default {
  title: 'TreeList',
  component: TreeList,
}

export const Base = (args) => <Content {...args} />
export const Background = (args) => <div className="ola-bg"><Content {...args} /></div>

function Content(...args) {
  return <TreeList {...args}>
    <TreeListOption>
      <a href="">
        <Issue title="This is the title of an issue" />
      </a>
    </TreeListOption>
    <TreeListOption>
      <a className="ola-bg-hover" href="">
        <Issue title="This is the title of an issue type pending" variant="pending" />
      </a>
    </TreeListOption>
    <TreeListOption>
      <a href="">
        <Issue title="This is the title of an issue" />
      </a>
    </TreeListOption>
    <TreeListOption>
      <a href="" className="ola-bg-hover"><Level /> Option 4 with level <Tag>and a tag</Tag></a>
    </TreeListOption>
    <TreeListOption>
      <a href="">Option 5 with link</a>
    </TreeListOption>
    <TreeListOption>
      <a href="">Option 6 with link</a>
    </TreeListOption>
    <TreeListOption>
      <a href="">
        <Page
          title="This is the title of a page"
          url="/path/of/the/page"
          icon={<Icon name="page" />}
        />
      </a>
      <TreeList>
        <TreeListOption>
          <a href="">
            <Issue size="small" title="This is the title of an issue" />
          </a>
        </TreeListOption>
        <TreeListOption>
          <a href="">
            <Issue size="small" title="This is the title of an issue" />
          </a>
        </TreeListOption>
        <TreeListOption>
          <a href="">
            <Issue size="small" title="This is the title of an issue" />
          </a>
        </TreeListOption>
      </TreeList>
    </TreeListOption>
    <TreeListOption variant="link">
      <div>
        <Issue title="This is the title of an issue" />
      </div>
    </TreeListOption>
  </TreeList>
}