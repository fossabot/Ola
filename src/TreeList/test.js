import React from 'react'
import TreeList from './'
import TreeListOption from './Option'
import renderer from 'react-test-renderer'

it('Default treeList', () => {
  const tree = renderer.create(<TreeList>
    <TreeListOption>Option 1</TreeListOption>
    <TreeListOption>Option 2</TreeListOption>
  </TreeList>).toJSON()
  expect(tree).toMatchSnapshot()
})
