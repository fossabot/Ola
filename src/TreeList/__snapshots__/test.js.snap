// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default treeList 1`] = `
<ul
  className="ola_treeList"
>
  <li
    className="ola_treeList-option"
  >
    Option 1
  </li>
  <li
    className="ola_treeList-option"
  >
    Option 2
  </li>
</ul>
`;
